daemon/cmd/fqdn.go

```diff
@@ -525,6 +527,43 @@ func (d *Daemon) notifyOnDNSMsg(lookupTime time.Time, ep *endpoint.Endpoint, epI
 	record.Log()
 
 	if msg.Response && msg.Rcode == dns.RcodeSuccess && len(responseIPs) > 0 {
+		// Create a critical section especially for when multiple DNS requests
+		// are in-flight for the same name (i.e. cilium.io).
+		//
+		// It is possible for the following race to occurr:
```

Comment:
nit: This comment declares that a particular race can occur. But the patch is changing the code to avoid this race. We should probably frame it in terms of "In the absence of such a lock, consider the following race condition: ..."